### PR TITLE
[Pfem] Removing tetgen submodule

### PR DESCRIPTION
**Pull request by josep-m-carbonell**
This is an attempt to remove tetgen submodule and adding automatic download instead. I don't know if it works in windows, I guess something different is needed. Maybe @roigcarlo can add something to not lose the advantages of having it as a submodule in multi-platform.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -69,3 +69,4 @@ kratos/includes/kratos_version.h
 *.bak
 
 applications/python_scripts/beam_profiles\.csv
+applications/DelaunayMeshingApplication/external_libraries/
```
Comment (philbucher) [inline .gitignore]: this should go to your local gitignore

Same as here: https://github.com/KratosMultiphysics/Kratos/tree/master/applications/ShapeOptimizationApplication/test_examples

Comment (roigcarlo) [reply .gitignore]: Didn't know you could have local ignores. In this case yep, that's better

Comment (josep-m-carbonell) [reply .gitignore]: ok

**Comment by roigcarlo** [comment]
Everything works for me but I think the `external_modules` folder should be deleted as well now that tetget is in external_libraires. Apart from that :+1:

**Comment by josep-m-carbonell** [comment]
I can not locate the external_modules folder, maybe it is in your local

**Review by roigcarlo — APPROVED** [review]
In this case 👍

**Pull request merged**